app/javascript/components/record-form/nav/component.spec.js

```diff
@@ -186,59 +179,39 @@ describe("<Nav />", () => {
     hasForms: true
   };
 
-  beforeEach(() => {
-    ({ component } = setupMountedComponent(Nav, props, initialState));
-  });
-
   it("renders a CloseIcon component />", () => {
-    expect(component.find(CloseIcon)).to.have.lengthOf(1);
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByTestId("close-icon")).toBeInTheDocument();
   });
 
   it("renders a Nav component />", () => {
-    expect(component.find(Nav)).to.have.lengthOf(1);
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByTestId("nav-list")).toBeInTheDocument();
   });
 
   it("renders a RecordInformation component />", () => {
-    expect(component.find(RecordInformation)).to.have.lengthOf(1);
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByText("forms.record_types.record_information")).toBeInTheDocument();
   });
 
   it("renders a Divider component />", () => {
-    expect(component.find(Divider)).to.have.lengthOf(1);
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByTestId("Divider")).toBeInTheDocument();
   });
 
   it("renders a NavGroup component from record information and another one from the others forms groups />", () => {
-    expect(component.find(NavGroup)).to.have.lengthOf(2);
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByRole('list')).toBeInTheDocument();
   });
 
   it("renders the NavGroup component for record information open", () => {
-    expect(component.find(NavGroup).first().props().open).to.equal("record_information");
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByText("forms.record_types.record_information")).toBeInTheDocument();
   });
 
   it("renders a ConditionalWrapper />", () => {
-    expect(component.find(ConditionalWrapper)).to.have.lengthOf(1);
-  });
-
-  it("should render valid props", () => {
-    const navProps = { ...component.find(Nav).props() };
-
-    expect(component.find(Nav)).to.have.lengthOf(1);
-    [
-      "firstTab",
-      "formNav",
-      "handleToggleNav",
-      "isNew",
-      "mobileDisplay",
-      "primeroModule",
-      "recordType",
-      "selectedForm",
-      "selectedRecord",
-      "toggleNav",
-      "hasForms"
-    ].forEach(property => {
-      expect(navProps).to.have.property(property);
-      delete navProps[property];
-    });
-    expect(navProps).to.be.empty;
+    mountedComponent(<Nav {...props} />, initialState);
+    expect(screen.getByTestId("ConditionalWrapper")).toBeInTheDocument();
   });
 
   describe("when is a new record", () => {
```

Comment:
From here on down, doesn't test what should be tested

Reply:
Written test cases to check if there is a new record